src/com/company/Main.java

```diff
@@ -102,14 +122,33 @@ public void traverseBook(BookNode book, List<BookNode> listOfBooks) { // inorder
 public class Main {
 
     static LibraryManagement library = new LibraryManagement();
-    static List<String> prompts = new ArrayList<>();
+    static BufferedWriter writer;
 
     public static void main(String[] args) {
+        try {
+            _readBookList();  //for reading from inputPS4.txt
+            triggerFunctionforPromptsPS4File(); //for reading from promptsPS4.txt
+            closeWriter();
+        } catch (Exception e) {
+            // finding the line number and method name of exception origin directly
+            System.out.println(e.getStackTrace()[0].getLineNumber() + "|" + e.getStackTrace()[0].getMethodName());
+            e.printStackTrace();
+        }
+    }
```

Comment:
if try catch is handled individually in all the function , then there is no need to handle in main. . in case of error there will be multiple outputs

Reply:
ok